### DISCUSSION TITLE
`useId` has incorrect TypeScript definition

### DISCUSSION THREAD

**Discussion by paya-cz** [quasarframework/quasar#17284] [General - Components / Directives / etc]
[Looking at the code](https://github.com/quasarframework/quasar/blob/dev/ui/src/composables/use-id/use-id.js), the composable directly returns a ref.

[The TypeScript definition](https://github.com/quasarframework/quasar/blob/dev/ui/types/composables.d.ts#L39) wraps it in an object `{ id: Ref<string> }`.

This generates TypeScript errors in my IDE because I am trying to bind the return value to `id` and `for` attributes, which expect a string type but instead the type is `{ id: Ref<string> }`.

**Comment by slovacekjan** [comment]
@paya-cz Hey Pavel, could you please reach out to me at the email address: slovacek@ofisly.com? We are very interested in potential cooperation. Unfortunately, I have not found any better solution for reaching you... Thank you!

Jan

**Comment by yusufkandemir** [comment]
Fixed by https://github.com/quasarframework/quasar/commit/b61b1dd64d7f30ad4fce32a020cd84ca2e66e9ed. The fix will be included in the next release.
